src/approaches/interactive_learning_approach.py

```diff
@@ -223,7 +225,12 @@ def _create_glib_interaction_strategy(
                            key=self._score_atom_set_frequency,
                            reverse=True)  # largest to smallest
         task_list = [Task(init, goal) for goal in goal_list]
-        task, act_policy = self._find_first_solvable(task_list)
+        try:
+            task, act_policy = self._find_first_solvable(task_list)
+        except ApproachFailure:
```

Comment:
what about approachtimeout?

Reply:
that gets caught within `_find_first_solvable`. this is only catching the last line of that method